packages/svelte/src/internal/client/runtime.js

```diff
@@ -1096,6 +1097,24 @@ export function pop(component) {
 	return component || /** @type {T} */ ({});
 }
 
+/**
+ *
+ * This is called from the inspect.
+ * Deeply traverse every item in the array with `deep_read` to register for inspect callback
+ * If the item implements INSPECT_SYMBOL, will use that instead
+ * @param {Array<any>} value
+ * @returns {void}
+ */
+function deep_read_inpect(value) {
+	for (const item of value) {
+		if (item && typeof item[INSPECT_SYMBOL] === 'function') {
```

Comment:
Shouldn't this be part of `deep_read` instead for where we check against the prototypes?

Reply:
i was thinking of doing that at first, but i wasn't sure if we should add more logic into the `deep_read`, which has use cases outside of inspect.
in those use cases, im not sure if we should check for the `INSPECT_SYMBOL`

Reply:
Valid point. Maybe leave it as it is then.